File: two-sum.py

# brute force
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

        return []
                    

# two-pass hash table
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}

        for i in range(len(nums)):
            comp = target - nums[i]
            if comp in numMap:
                return [i, numMap[comp]]
            numMap[nums[i]] = i

        return []
                    

        

File: two-sum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

// two-pass hash table
class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int, int> numMap;

            for (int i{}; i < nums.size(); i++) {
                numMap[nums[i]] = i;
            }

            for (int i{}; i < nums.size(); i++) {
                int comp = nums[i] - target;
                if (numMap.count(comp) && comp != i) return {i, numMap[comp]};
            }
        }
    };